org-ql.el

```diff
@@ -448,6 +452,29 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                  org-ql-tags-cache))
       (puthash position all-tags tags-cache))))
 
+(defun org-ql--expand-tag-groups (tags &optional exclude-groups)
+  (if (eq tags 'org-ql-nil)
+      'org-ql-nil
+    (let ((exclude-groups (append tags exclude-groups))
+          result)
+      (let (group-tags)
+        (dolist (tag tags)
+          (dolist (groups org-tag-groups-alist)
```

Comment:
Is `org-tag-groups-alist` the right or best variable to use?  Looking at `org.el`, I see:

```el
(defvar-local org-tag-groups-alist nil)
...
(defvar-local org-current-tag-alist nil
  "Alist of all tag groups in current buffer.
This variable takes into consideration `org-tag-alist',
`org-tag-persistent-alist' and TAGS keywords in the buffer.")
```
It's a potential red flag that that variable isn't documented and the other one is, and the docstring of the other one seems to suggest it might be more correct.  

(The lack of clarity on this may also suggest that supporting this feature could get a little messy.  We'd better be sure to write good tests for this feature.)

Reply:
I didn't notice that. Thankfully, instead of `org-tag-groups-alist`, we can use `(org-tag-alist-to-groups org-current-tag-alist)` which appears less likely to cause maintenance issues. Addressed in the new commit.